### PR TITLE
Introduce separate backend naming scheme for RXLB Ingress

### PR DESCRIPTION
**Pull request by panslava**
- This is required to share the same service between RXLB and ILB. Frontend resources already have separate names (they depend on the ingress)

/assign @spencerhance

### CODE REVIEW

**Comment by spencerhance** [comment]
generally LGTM but would like another pair of eyes due to inherent risk in modifying namers

**Comment by panslava** [comment]
there might be a bug

/hold

File: pkg/utils/serviceport.go [hunk 1]
```diff
@@ -85,13 +85,22 @@ func (sp *ServicePort) GetDescription() Description {
 
 // BackendName returns the name of the backend which would be used for this ServicePort.
 func (sp *ServicePort) BackendName() string {
-	if sp.NEGEnabled {
-		return sp.BackendNamer.NEG(sp.ID.Service.Namespace, sp.ID.Service.Name, sp.Port)
-	} else if sp.VMIPNEGEnabled || sp.L4RBSEnabled {
+	if sp.L7XLBRegionalEnabled {
```
Comment (swetharepakula) [inline pkg/utils/serviceport.go:88]: can you add a unit test to test this condition?

File: pkg/utils/serviceport.go [hunk 2]
```diff
@@ -85,13 +85,22 @@ func (sp *ServicePort) GetDescription() Description {
...
+		return sp.BackendNamer.RXLBBackendName(sp.ID.Service.Namespace, sp.ID.Service.Name, sp.Port)
+	} else if sp.NEGEnabled || sp.VMIPNEGEnabled || sp.L4RBSEnabled {
+		// L4, and ILB, GXLB Ingresses are using NEG Name for all backend resources.
+		return sp.NEGName()
+	}
+	return sp.BackendNamer.IGBackend(sp.NodePort)
+}
+
+func (sp *ServicePort) NEGName() string {
```
Comment (swetharepakula) [inline pkg/utils/serviceport.go:97]: can we add unit tests for NegName since it is now exposed to be used by itself?

**Review by swetharepakula — COMMENTED** [review]
This does maintain the status quo for not L4 RXLB. 

I think we need a couple more unit tests that test the conditional logic in the two naming functions (BackendName & NEGName).

File: pkg/utils/serviceport.go (hunk 2 above)
Comment (panslava) [reply pkg/utils/serviceport.go:97]: Added unit tests for NEGName

File: pkg/utils/serviceport.go (hunk 1 above)
Comment (panslava) [reply pkg/utils/serviceport.go:88]: Added unit tests for BackendName

**Review by swetharepakula — COMMENTED** [review]
Thanks Slavik!

/lgtm
/approve

**Pull request merged**